src/ckeditor.js

```diff
@@ -12,9 +14,56 @@
  * @singleton
  */
 
-CKEDITOR.define( function() {
+CKEDITOR.define( [ 'editor', 'mvc/collection', 'promise' ], function( Editor, Collection, Promise ) {
 	var CKEDITOR = {
 		/**
+		 * A collection containing all editor instances created.
+		 */
+		instances: new Collection(),
+
+		/**
+		 * Creates an editor instance for the provided DOM element.
+		 *
+		 * The creation of editor instances is an asynchronous operation, therefore a promise is returned by this
+		 * method.
+		 *
+		 *     CKEDITOR.create( '#content' );
+		 *
+		 *     CKEDITOR.create( '#content' ).then( function( editor ) {
+		 *         // Manipulate "editor" here.
+		 *     } );
+		 *
+		 * @param {String|Object} element An element selector or a DOM element, which will be the source for the
+		 * created instance.
+		 * @returns {Promise} A promise, which will be fulfilled with the created editor.
+		 */
+		create: function( element ) {
+			var that = this;
+
+			return new Promise( function( resolve, reject ) {
+				// If a query selector has been passed, transform it into a real element.
+				if ( typeof element == 'string' ) {
+					element = document.querySelector( element );
```

Comment:
The cases that makes me sad about the array approach are these:

```
// When pointing to a specific ID... for a single element
CKEDITOR.create( '#content' ).then( function( editors ) {
    // Why not just the editor that I created :(
    var editor = editors[ 0 ];
} );

// Even worst, when passing a single element.
var el = document.getElementById( 'content' );
CKEDITOR.create( el ).then( function( editors ) {
    // Why not just the editor that I created :(
    var editor = editors[ 0 ];
} );
```

Those are the most common cases because usually developers control each editor creation instead of creating a bunch of them.

For now, I would like to wait with this features, creating just a single editor per call.